python_modules/dagster/dagster_tests/asset_defs_tests/test_materialize.py

```diff
@@ -151,6 +152,30 @@ def the_asset(the_source):
         assert result.output_for_node("the_asset") == 6
 
 
+def test_materialize_asset_specs():
+    class MyIOManager(IOManager):
+        def handle_output(self, context, obj):
+            pass
+
+        def load_input(self, context):
+            return 5
+
+    the_source = AssetSpec(
+        key=AssetKey(["the_source"]), metadata={"dagster/io_manager_key": "my_io_manager"}
```

Comment:
this likely merits a discussion

Reply:
Some background here: https://github.com/dagster-io/internal/discussions/10536#discussioncomment-10036256